### ISSUE TITLE
ANSI-C quotes applied to strings without ANSI-C escape sequences

### ISSUE THREAD

**Issue by GeorgeHahn** [allenap/shell-quote#8]
Hi, thanks for the great library!

I've noticed that some strings are unnecessarily quoted for ANSI-C escape sequences. This seems to happen on strings that contain an equal sign. For example, the string `--arg=var` will be quoted as `$'--arg=var'`. This is valid and doesn't cause any issues, but it does appear to be unnecessary.

I wasn't familiar with this style of quoting. It is covered in the bash docs here:

> 3.1.2.4 ANSI-C Quoting
> 
> Character sequences of the form $’string’ are treated as a special kind of single quotes. The sequence expands to string, with backslash-escaped characters in string replaced as specified by the ANSI C standard. Backslash escape sequences, if present, are decoded as follows: 
>
> \a
>    alert (bell) 
>
> \b
>    backspace 
> ...
> The expanded result is single-quoted, as if the dollar sign had not been present. 

([source](https://www.gnu.org/software/bash/manual/bash.html#ANSI_002dC-Quoting-1))

**Comment by allenap** [comment]
Yeah, it is a bit weird/confusing. I _think_ I chose this style because of Unicode support:

> `\uHHHH`
> the Unicode (ISO/IEC 10646) character whose value is the hexadecimal value HHHH (one to four hex digits)
>
> `\UHHHHHHHH`
> the Unicode (ISO/IEC 10646) character whose value is the hexadecimal value HHHHHHHH (one to eight hex digits)

Perhaps the library could optimise so that it only uses `$'...'` style when it needs this support 🤔

**Comment by fenollp** [comment]
Isn't the leading dash the cause for this quoting, and not the `=`?

**Comment by allenap** [comment]
> Isn't the leading dash the cause for this quoting, and not the `=`?

The Bash quoting code classifies (well, the private `ascii` and `utf8` modules do) hyphens / `-` as printable-inert ([bytes](https://github.com/allenap/shell-quote/blob/b5365b3a8b17c23c09f168f0c15b0c3214facfdc/src/ascii.rs#L57), [text](https://github.com/allenap/shell-quote/blob/b5365b3a8b17c23c09f168f0c15b0c3214facfdc/src/utf8.rs#L56)), whereas equals / `=` is classified as printable ([bytes](https://github.com/allenap/shell-quote/blob/b5365b3a8b17c23c09f168f0c15b0c3214facfdc/src/ascii.rs#L62), [text](https://github.com/allenap/shell-quote/blob/b5365b3a8b17c23c09f168f0c15b0c3214facfdc/src/utf8.rs#L61)), meaning they _might_ be interpreted specially in the shell. The Bash `escape_prepare` methods ([bytes](https://github.com/allenap/shell-quote/blob/b5365b3a8b17c23c09f168f0c15b0c3214facfdc/src/bash.rs#L199-L210), [text](https://github.com/allenap/shell-quote/blob/b5365b3a8b17c23c09f168f0c15b0c3214facfdc/src/bash.rs#L252-L263)) will only omit quoting when every character is inert.
